Improve download file helpers

lennym opened the pull request:
Allow any download link to be used without it needing to be part of a download header or page. Attaches a download method onto an element so you can download the target of any download link:

```js
const content = browser.$('a=Download').download('pdf');
```

Also parses csv data into structured objects:

```js
const csv = browser.$('a=Download active licence list').download('csv');
assert.ok(csv.every(row => row.status === 'active'));
```

lennym committed:
Improve download file helpers

Allow any download link to be used without it needing to be part of a download header or page. Attaches a download method onto an element so you can download the target of any download link:

```js
const content = browser.$('a=Download').download('pdf');
```

Also parses csv data into structured objects:

```js
const csv = browser.$('a=Download active licence list').download('csv');
assert.ok(csv.every(row => row.status === 'active'));
```

wheelsandcogs commented on helpers/download-file.js:
```diff
@@ -1,8 +1,44 @@
+        switch (type) {
+          case 'pdf':
+            return parsePDF(data).then(pdf => pdf.text.replace(/\s/g, ' '));
+          case 'word':
```
```suggestion
          case 'word':
            return data.toString('utf8');
```

lennym committed:
Automatically detect file type from response header

If the type is not defined then use the response header to work out what the type should be and parse accordingly.
